Training_node/first_project/app/controllers/users.js

```diff
@@ -0,0 +1,62 @@
+const User = require('../models/').user,
+  validator = require('validator'),
+  Hash = require('../services/bcrypt'),
+  errors = require('../errors'),
+  logger = require('../logger');
+
+const validateRestrictions = (email, password) => {
+  if (!(validator.isEmail(email) && validator.contains(email, '@wolox.com.ar'))) {
+    return { success: false, reason: 'The email must be valid and pertain Wolox' };
+  } else {
+    if (!(validator.isAlphanumeric(password) && password.length > 8)) {
+      return { success: false, reason: 'The password must be alphanumeric and length greather than 8' };
+    } else {
+      return { success: true };
+    }
+  }
+};
+
+const validate = (firstName, lastName, password, email) => {
+  return !firstName || !lastName || !password || !email
+    ? { success: false, reason: 'The fields first name, last name, password and email are required' }
+    : { success: true };
+};
+
+exports.create = (req, res, next) => {
+  const saltRounds = 10;
+  const user = req.body
+    ? {
+        firstName: req.body.firstName,
+        lastName: req.body.lastName,
+        password: req.body.password,
+        email: req.body.email
+      }
+    : {};
+  let validation = validate(user.firstName, user.lastName, user.password, user.email);
```

Comment:
why let? it should be const

Reply:
@gonziesc because the variable validation is assigned again later

Reply:
do not use the same var then